dbt/include/athena/macros/materializations/models/table/create_table_as.sql

```diff
@@ -36,9 +36,7 @@
     {%- endif -%}
   {%- endif %}
 
-{%- if table_type != 'iceberg' -%}
-    {% do adapter.prune_s3_table_location(location) %}
-{%- endif -%}
+  {% do adapter.delete_from_s3(location) %}
```

Comment:
@Jrmyy note this, generally we drop the old relation before calling the create_table_as, so there shouldbn't be any conflicnt.
This is a safe guard also for iceberg, in some cases (when using vacuuming), a drop iceberg table left the path dirty, this change ensure that the path is always clean before writing.